### DISCUSSION TITLE
Cutting off text so that it fits in one line in the header

### DISCUSSION THREAD

**Discussion by adrfantini** [typst/typst#4076] [General]
I want to have a header with a line of text + an image. If the line of text is too long, it should cutoff and display ellipsis.

My current MRE implementation of this is this, but it's quite ugly: is there a better way? Thanks!

```typst
#let mybox = box(baseline: 20%, rect(height: 1em))

#set page(
  header: layout(size => context [
    #let boxwidth = measure(mybox).width
    #let n = 30
    #let text = "Header text: " + lorem(n)
    #let textwidth = measure(text).width // a reasonable number
    #while boxwidth + textwidth > size.width {
      n = n - 1
      text = "Header text: " + lorem(n) + "..."
      textwidth = measure(text).width
    }
    #text
    #h(1fr)
    #mybox
  ])
)
```

**Comment by adrfantini** [comment]
About 1 year later I think this is still a significant issue. I did not find a better way to dynamically truncate a text based on the available space, without going to new lines.

**Comment by Joel-Singh** [comment]
There is https://typst.app/universe/package/chomp/ which does exactly this.

**Reply by adrfantini** [reply]
Thanks, that's cool! I'm unsure if this should be built-in in Typst to be fair or if not, and I should close the issue

**Comment by Joel-Singh** [comment]
For others seeing this discussion, the solution I settled on for my use case of hydra headers is here: https://forum.typst.app/t/truncating-hydra-headers-to-a-single-line/8152/4?u=apple
